transform/flatten_mangler_test.go

```diff
@@ -269,7 +302,103 @@ func TestFlattenMangler(t *testing.T) {
 					DayTripper: &b2,
 				}
 
-				assert.Equal(t, &s, i)
+				assert.EqualValues(t, &s, i)
+			},
+		},
+		{
+			name:       "Embedded struct without tag",
+			testStruct: efg,
+			modify: func(t testing.TB, val reflect.Value) {
+				expectedTags := []string{
+					`dials:"ConfigField_Name"`,
+					`dials:"ConfigField_Location"`,
+					`dials:"ConfigField_Coordinates"`,
+					`dials:"ConfigField_AnotherField"`,
+				}
+
+				expectedNames := []string{
+					"ConfigFieldName",
+					"ConfigFieldLocation",
+					"ConfigFieldCoordinates",
+					"ConfigFieldAnotherField",
+				}
+
+				vtype := val.Type()
+				for i := 0; i < vtype.NumField(); i++ {
+					assert.EqualValues(t, expectedTags[i], vtype.Field(i).Tag)
+					assert.EqualValues(t, expectedNames[i], vtype.Field(i).Name)
+				}
+
+				s1 := "test"
+				s2 := "here"
+				i1 := 64
+				i2 := 42
+
+				val.Field(0).Set(reflect.ValueOf(&s1))
+				val.Field(1).Set(reflect.ValueOf(&s2))
+				val.Field(2).Set(reflect.ValueOf(&i1))
+				val.Field(3).Set(reflect.ValueOf(&i2))
+			},
+			assertion: func(t testing.TB, i interface{}) {
+				// embedded fields are hard to compare with defined structs because
+				// they are named but the Anonymous field is set to false. So use
+				// JSON marshaling/unmarshalling to compare values
+
+				b, err := json.Marshal(i)
+				assert.NoError(t, err)
+
+				var actual embeddedFooBar
+				err = json.Unmarshal(b, &actual)
+				assert.NoError(t, err)
+				assert.Equal(t, efg, actual)
+			},
+		},
+		{
+			name:       "Embedded struct with tag",
+			testStruct: efgt,
+			modify: func(t testing.TB, val reflect.Value) {
+				expectedTags := []string{
+					`dials:"ConfigField_Name"`,
+					`dials:"ConfigField_embeddedFoo_Location"`,
+					`dials:"ConfigField_embeddedFoo_Coordinates"`,
+					`dials:"ConfigField_AnotherField"`,
+				}
+				expectedNames := []string{
+					"ConfigFieldName",
+					"ConfigFieldLocation",
+					"ConfigFieldCoordinates",
+					"ConfigFieldAnotherField",
+				}
+
+				vtype := val.Type()
+				for i := 0; i < vtype.NumField(); i++ {
+					assert.EqualValues(t, expectedTags[i], vtype.Field(i).Tag)
+					assert.EqualValues(t, expectedNames[i], vtype.Field(i).Name)
+				}
+
+				s1 := "test"
+				s2 := "here"
+				i1 := 64
+				i2 := 42
+
+				val.Field(0).Set(reflect.ValueOf(&s1))
+				val.Field(1).Set(reflect.ValueOf(&s2))
+				val.Field(2).Set(reflect.ValueOf(&i1))
+				val.Field(3).Set(reflect.ValueOf(&i2))
+			},
+			assertion: func(t testing.TB, i interface{}) {
+				// assert.EqualValues doesn't work here with the embedded structs
+				// like it does for nested structs since the values are different
+				// with Anonymous set to true for embedded fields. So using JSON
+				// marshalling to ensure that the values are populated correctly
+				b, err := json.Marshal(i)
+				assert.NoError(t, err)
```

Comment:
perhaps `s/assert/require/` here so we don't get failures further down.

Reply:
Good call. Yes require would be better. Fixed